### PR TITLE
Update dependencies and switch to yarn

### PR DESCRIPTION
**Pull request by aburgel**
/domain @chrislopresto @asullivan210 @samandmoore @jmileham 
/no-platform

The test config got a little stale and no longer seems to run with the latest phantomjs. I've updated most of our dev dependencies to the latest, switched the tests to run with headless Chrome, and switched to using yarn instead of npm and bower.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,5 +52,5 @@
     "feature flagging",
     "test track"
   ],
-  "license": "LICENSE"
+  "license": "MIT"
```
Comment (samandmoore) [inline package.json]: Doesn’t matter much but LICENSE was meant to tell folks to read the LICENSE file. Maybe that’s not valid in yarn?

**Comment by samandmoore** [comment]
<<domainlgtm

File: package.json (hunk 1 above)
Comment (aburgel) [reply package.json]: Its not valid in NPM either, it produces a warning, same as in yarn.

**Comment by samandmoore** [comment]
Sgtm change away

**Pull request merged**